cmd/thv/app/list.go

```diff
@@ -108,10 +109,14 @@ func printMCPServersOutput(workloadList []core.Workload) error {
 	mcpServers := make(map[string]map[string]string)
 
 	for _, c := range workloadList {
+		// Determine the effective transport type to show to users
```

Comment:
We probably should do this in the workload manager so that the behaviour is consistent across UI and API.